frontend/views/components/AvatarUser.vue

```diff
@@ -2,47 +2,59 @@
   avatar(
     :src='pictureURL'
     :alt='alt'
+    :size='size'
   )
 </template>
 
 <script>
+import { mapGetters } from 'vuex'
 import sbp from '~/shared/sbp.js'
 import Avatar from '@components/Avatar.vue'
 
 export default {
-  name: 'UserImage',
+  name: 'AvatarUser',
   components: { Avatar },
   props: {
     username: String,
     alt: {
       type: String,
       default: ''
+    },
+    size: {
+      type: String,
+      default: 'md',
+      validator: (value) => ['xs', 'sm', 'md', 'lg', 'xl'].includes(value)
+    }
+  },
+  data () {
+    return {
+      ephemeral: {
+        url: null
+      }
     }
   },
   async mounted () {
     if (!this.profile) {
+      console.debug(`Looking for ${this.username} profile picture`)
       const userContractId = await sbp('namespace/lookup', this.username)
       if (!userContractId) {
-        console.error(`UserImage: ${this.username} doesn't exist!`)
+        console.error(`AvatarUser: ${this.username} doesn't exist!`)
         return
       }
       const state = await sbp('state/latestContractState', userContractId)
       this.ephemeral.url = state.attributes.picture
     }
   },
   computed: {
-    profile () {
-      return this.$store.getters.globalProfile(this.username)
+    ...mapGetters([
+      'ourGroupProfile'
+    ]),
+    profilePicture () {
+      const profile = this.ourGroupProfile && this.$store.state[this.ourGroupProfile.contractID]
+      return profile && profile.attributes && profile.attributes.picture
     },
     pictureURL () {
-      return this.profile ? this.profile.picture : this.ephemeral.url
-    }
-  },
-  data () {
-    return {
-      ephemeral: {
-        url: null
-      }
+      return this.profilePicture || this.ephemeral.url
```

Comment:
I'm seeing two issues with this change:

1. By renaming `profile` computed property to `profilePicture`, you're introducing a bug on line 37 which checks for `this.profile`
2. I do not understand why `profile` was modified in the first place, as it was doing a good job and in fewer lines...

Reply:
1. Fixed
2. Yes, this isn't correct, but the old solution isn't fully correct either.  `globalProfile` isn't truly getting the global profile. It only searches for profiles in the current group. If we use this getter on a chat when talking with someone that's not part of our group, that globalProfile will return undefined. But that's a problem to fix later... I reverted to `globalProfile()` and later we can improve `globalProfile` to work properly.